### ISSUE TITLE
[FEAT] Optional text on Sankey links

### ISSUE THREAD

**Issue by emmanuelle** [plotly/plotly.js#4746]
At the moment Sankey _nodes_ display their labels as text, but for the links one can only have text on hover. It would be great to have an option to display text inside (or just above) the links.

Related issue: add optional text to histogram and to heatmap traces (which would remove the need for annotated heatmap figure factory).

**Comment by antoinerg** [comment]
Do you have any suggestions about how the users would specify this? Do you need a new attribute?

cc @nicolaskruchten

**Comment by emmanuelle** [comment]
maybe just a `texttemplate` which would be empty by default? And/or `textinfo` as in `pie` or `funnel`?

**Comment by kenakofer** [comment]
Yes please. It's strange that in the case of Sankey links, specifying a label is actually just specifying a hovertemplate.

**Comment by wattcommunity** [comment]
OK a Sankey is a visualisation tool, but come on, seeing the magnitude of the pipes in a png export is like, why wouldn't you make this doable! Sorry to gripe; but other Sankey tools are pants and so plotly's offer is a last gasp saloon.

**Comment by ccpjulia** [comment]
I can see that there is a `label` attribute for `link`, but adding labels to it does nothing. And agreed being able to specify `texttemplate` would also be great.

**Comment by iyliahutta** [comment]
is this in development or will be anytime soon? would really love to have this feature and i doubt i'm the only one.

**Comment by fstricker** [comment]
Is there any news on this issue? I find it strange that there is no way of showing persistent link texts regardless of hover states.

**Comment by wf-r** [comment]
I would be willing to work on a PR for this, if there is interest.

## Motivation
 
Sankey links currently expose their information (name via `link.label`, plus value) only through the hover tooltip. There is no way to display link text permanently on the diagram, the way node labels are always drawn. This forces users to either accept hover-only interaction (useless for static export/PNG/print) or build fragile client-side SVG overlays.
 
There is some community interest in exactly this - see this issue.

## Implementation options
 
### Option A — reuse `link.label` as the on/off switch
 
Render `link.label` when it is set.

This would need no new attributes, however one would only be able to show the label string, no value or custom text. It would also silently change behavior for existing figures that set `link.label` purely for hover - i.e. **not backward compatible**.

### Option B — introduce `link.textinfo`
 
A flaglist string: any combination of `"label"`, `"value"` or other flags joined with `"+"`, or `"none"`.
 
- **Pro:** explicit opt-in (default `"none"` → no behavior change); users choose which pieces of information to show; consistent with the `textinfo` pattern already established in traces such as [`funnel`](https://plotly.com/javascript/reference/funnel/#funnel-textinfo) and [`pie`](https://plotly.com/javascript/reference/pie/#pie-textinfo), so the attribute semantics are familiar to users and reusable in code.
- **Con:** fixed formatting/ordering; no full control over the rendered string.

### Option C — `link.textinfo` (as in B) plus `link.texttemplate`
 
`texttemplate` is a string or array of strings: a template for the rendered text (e.g. `"%{label}: %{value}"`), overriding `textinfo` when set; supports the same variables as the link `hovertemplate` (`value`, `label`, `source`, `target`, `customdata`, …).
 
- **Pro:** full control, per-link templating, and still a clean opt-in. The `textinfo` + `texttemplate` duo is exactly the pattern `funnel` and `pie` already implement (see [`pie.texttemplate`](https://plotly.com/javascript/reference/pie/#pie-texttemplate) and [`funnel.texttemplate`](https://plotly.com/javascript/reference/funnel/#funnel-texttemplate)), so we can follow — and largely reuse — their template machinery rather than inventing anything new.
- **Con:** the most code (template parsing, array handling), but reuses existing template machinery.

## Backward compatibility
 
- **Option A:** not backward compatible — existing figures that set `link.label` for hover would suddenly render on-diagram text.
- **Options B/C:** fully backward compatible. New attributes default to `textinfo: "none"` (and `texttemplate: ""`), so existing figures render identically; link text is strictly opt-in. No changes to node labels, hover, or existing baselines.

## Key implementation points
 
- Render one `<text>` per link inside the `.sankey` group at the link midpoint (`(source.x1 + target.x0) / 2`, `(y0 + y1) / 2`), so the existing group-level orientation transform applies.
- Apply the same counter-transform as node labels (`flipText`) so text stays upright in all orientations (`h`/`v`).
- Reposition the labels in `updateShapes` so they follow live node drags (the underlying `y0`/`y1` are already recomputed there).
- For circular links the analytic midpoint is inaccurate; use `path.getPointAtLength(totalLength / 2)` for a robust anchor.
- Respect a new `link.textfont` (mirroring `node`/trace `textfont`) and `valueformat`/`valuesuffix` for the value component.

**Comment by wf-r** [comment]
An example for link labels with `link.textinfo` and `link.texttemplate` would be this:

<img width="1838" height="663" alt="Image" src="https://github.com/user-attachments/assets/ec658206-fac7-453d-be8d-e287025d896a" />

An example for the configuration with `link.valueformat`, `link.valuesuffix` and `link.textfont` would be this:

<img width="1835" height="333" alt="Image" src="https://github.com/user-attachments/assets/82f070c9-5127-4d84-8bb7-ce967425659d" />

**Comment by robertclaus** [comment]
Thanks for the PR @wf-r ! @camdecoster had a chance to review if you are still looking into this!
